### PR TITLE
feat: add agenda page for display the user events

### PR DESCRIPTION
**Pull request by zohary89**
This feature add a new agenda page for displaying the user events.
- Add agenda page route and html
- Add temporary agenda button to the header in base.html (until future changes)
- Add start date and end date to the event model
- Add tests
- Add basic css (should be upgrated later according on the web design)

### CODE REVIEW

**Comment by yammesicka** [comment]
Please change the merge request to merge into PythonFreeCourse:develop

File: app/internal/agenda_events.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+from datetime import date, datetime, timedelta
+from typing import Optional
+
+from app.database.models import Event
+from app.database.database import SessionLocal
+
+from sqlalchemy.exc import SQLAlchemyError
+
+
+def get_events_per_dates(session: SessionLocal, user_id: int, start: Optional[date], end: Optional[date]) -> list:
```
Comment (yammesicka) [inline app/internal/agenda_events.py:10]: Consider changing `list` to `List[DataType]`, where DataType is the type of the variables in the list.

File: app/internal/agenda_events.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    """Read function from the db. Return a list of all the user events between
+    the relevant dates."""
+    if start > end:
+        return []
+    try:
+        events = (
+            session.query(Event).filter(Event.owner_id == user_id)
+            .filter((Event.start >= start) & (Event.start <= end + timedelta(days=1)))
```
Comment (yammesicka) [inline app/internal/agenda_events.py:18]: Maybe `filter(start <= Event.start <= end + timedelta(days=1))`?

File: app/internal/agenda_events.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+            .order_by(Event.start).all()
+            )
+    except SQLAlchemyError:
+        return []
+    else:
+        return events
+
+
+def calc_dates_range_for_agenda(start: Optional[date], end: Optional[date], days: Optional[int]) -> (date, date):
```
Comment (yammesicka) [inline app/internal/agenda_events.py:27]: This line is a bit too long, split it :) (See flake8 output)

File: app/internal/agenda_events.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    if days is not None:
+        start = date.today()
+        end = start + timedelta(days=days)
+    elif start is None or end is None:
+        start = date.today()
+        end = date.today()
+    return start, end
+
+
+def get_time_delta_string(start_date: date, end_date: date) -> str:
```
Comment (yammesicka) [inline app/internal/agenda_events.py:38]: You might miss lots of edge cases. Try to use `arrow` instead.

File: app/main.py [hunk 5]
```diff
@@ -30,3 +42,32 @@ def profile(request: Request):
         "username": current_username,
         "events": upcouming_events
     })
+
+
+@app.get("/agenda")
```
Comment (yammesicka) [inline app/main.py:47]: This should be on routing/agenda.py

File: app/main.py [hunk 6]
```diff
@@ -30,3 +42,32 @@ def profile(request: Request):
...
+    request: Request,
+    start_date: Optional[date]=None,
+    end_date: Optional[date]=None,
+    days: Optional[int]=None) -> Jinja2Templates:
+    """Route for the agenda page, using dates range or exact amount of days."""
+    
+    user_id = 1   # there is no user session yet, so I use user id- 1.
+    start_date, end_date = agenda_events.calc_dates_range_for_agenda(start_date, end_date, days)
+
+    db_session = SessionLocal()
```
Comment (yammesicka) [inline app/main.py:58]: Read https://fastapi.tiangolo.com/tutorial/sql-databases/ to learn about handling db in fastapi

File: tests/test_agenda_route.py [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+        event = Event(title=title, content=content, start=start, end=end, owner_id=owner_id)
+        session.add(event)
+        session.commit()
+
+    @staticmethod
+    def create_data_user_1(session):
+        user = TestAgenda.create_user_1(session)
+        base_date = TestAgenda.base_today_date()
+        # Today event
+        TestAgenda.add_event(session, "event 1", "...", base_date + timedelta(hours=7), base_date + timedelta(hours=9), user.id)
```
Comment (yammesicka) [inline tests/test_agenda_route.py:44]: Maybe you should use parameterize? :)

**Review by yammesicka — CHANGES_REQUESTED** [review]
Good job! Added some comments :)

File: tests/test_agenda_route.py (hunk 7 above)
Comment (zohary89) [reply tests/test_agenda_route.py:44]: I'm not sure parameterize suitable here, each test is for specific usage of the url, and each test needs some events to check which of them show up and which not.
therefore, I built this helper function for adding a group of 6 events for use in all the tests.

File: app/templates/agenda.html [hunk 8]
```diff
@@ -0,0 +1,52 @@
+{% extends "base.html" %}
+
```
Comment (ivarshav) [inline app/templates/agenda.html:2]: I think it could be nice to add screenshots to your PR description.

File: tests/test_agenda_internal.py [hunk 9]
```diff
@@ -0,0 +1,33 @@
+from datetime import datetime
+
+from app.internal import agenda_events
+import pytest
+
+start = datetime(2021, 11, 1, 8, 00, 00)
```
Comment (ivarshav) [inline tests/test_agenda_internal.py:6]: Are they CONSTS? If so, probably should be CAPITAL.

File: tests/test_agenda_route.py [hunk 10]
```diff
@@ -0,0 +1,139 @@
+from datetime import date, datetime, timedelta
+from app.database.models import User, Event
+
+
+HTTP_OK = 200
```
Comment (ivarshav) [inline tests/test_agenda_route.py:5]: Can use starlette.status instead.

File: tests/test_agenda_route.py [hunk 11]
```diff
@@ -0,0 +1,139 @@
...
+
+    @staticmethod
+    def create_user_2(session):
+        user2 = User(username='dani', email='bb.aa@aa.com', password='12345')
+        session.add(user2)
+        session.commit()
+        return user2
+
+    @staticmethod
+    def add_event(session, title, content, start, end, owner_id):
```
Comment (ivarshav) [inline tests/test_agenda_route.py:34]: Consider call it just event.

**Pull request merged**